### PR TITLE
 Split ClientDeviceInformation in config and credentials 

### PR DESCRIPTION
**Pull request by danielpassos**
Related issues:
* [AGIOS-679](https://issues.jboss.org/browse/AGIOS-679)

### CODE REVIEW

File: modules/push/AgsPush.swift [hunk 1]
```diff
@@ -20,19 +30,73 @@ public class AgsPush {
...
+     - parameter credentials: Credentials used to register the device
+
+     - parameter success: A block object to be executed when the registration operation finishes successfully. This block has no return value.
+
+     - parameter failure: A block object to be executed when the registration operation finishes unsuccessfully.This block has no return value and takes one argument: The `NSError` object describing the error that occurred during the registration process.
+    */
+    public func register(_ deviceToken: Data,
+                  _ config: UnifiedPushConfig,
+                  _ credentials: UnifiedPushCredentials,
+                  success: (() -> Void)!,
```
Comment (wtrocki) [inline modules/push/AgsPush.swift]: Remove exclamation marks

File: modules/push/AgsPush.swift [hunk 2]
```diff
@@ -20,19 +30,73 @@ public class AgsPush {
...
+        let basicAuthCredentials: Data! = "\(credentials.variantID):\(credentials.variantSecret)".data(using: String.Encoding.utf8)
+        let base64Encoded = basicAuthCredentials.base64EncodedString(options: NSData.Base64EncodingOptions(rawValue: 0))
+        return ["Authorization": "Basic \(base64Encoded)"]
+    }
+
+    // Helper to transform the Data-based token into a (useful) String:
+    fileprivate func convertToString(_ deviceToken: Data?) -> String? {
+        if let token = (deviceToken as NSData?)?.description {
+            return token.replacingOccurrences(of: "<", with: "")
+                    .replacingOccurrences(of: ">", with: "")
```
Comment (wtrocki) [inline modules/push/AgsPush.swift]: Question: I copied that code. Is that still necessary? I tested that and there are no characters like this in token, but were not sure why this was added

**Comment by wtrocki** [comment]
Have no phone. Going to verify that later

File: modules/push/AgsPush.swift (hunk 2 above)
Comment (wtrocki) [reply modules/push/AgsPush.swift]: So I have added that to avoid some compat issues.

Comment (danielpassos) [reply modules/push/AgsPush.swift]: We can test without it and remove if still working!

File: modules/push/AgsPush.swift (hunk 1 above)
Comment (danielpassos) [reply modules/push/AgsPush.swift]: I'm confused here. If I remove it I will need to do

```swift
success: @escaping (() -> Void),
failure: @escaping ((NSError) -> Void))
```

Comment (wtrocki) [reply modules/push/AgsPush.swift]: Yeah. We need escaping. Sorry for not being clear.

File: example/AeroGearSdkExampleTests/push/DeviceRegistrationTests.swift [hunk 3]
```diff
@@ -1,89 +0,0 @@
-import AGSPush
```
Comment (wtrocki) [inline example/AeroGearSdkExampleTests/push/DeviceRegistrationTests.swift]: Let's restore this tests. Removing unit tests is not the best way to go. Let's adjust them in the future.

Comment (danielpassos) [reply example/AeroGearSdkExampleTests/push/DeviceRegistrationTests.swift]: Done in https://github.com/aerogear/aerogear-ios-sdk/pull/73

**Pull request merged**